File: backend/app/schemas/notification.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class NotificationCreate(BaseModel):
    event_id: Optional[int] = None
    recipient_type: str
    recipient_id: Optional[int] = None
    notification_type: str
    message: str
    sender: str

class NotificationResponse(BaseModel):
    id: int
    event_id: Optional[int] = None
    recipient_type: str
    recipient_id: Optional[int] = None
    notification_type: str
    message: str
    sender: str
    is_read: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/notification.ts
export interface Notification {
  id: number;
  event_id?: number;
  recipient_type: string;
  recipient_id?: number;
  notification_type: string;
  message: string;
  sender: string;
  is_read: boolean;
  created_at: string;
}

export interface NotificationCreate {
  event_id?: number;
  recipient_type: string;
  recipient_id?: number;
  notification_type: string;
  message: string;
  sender: string;
}